### DISCUSSION TITLE
Deleting a bucket with immutable objects

### DISCUSSION THREAD

**Discussion by Jazz9** [minio/minio#17009] [Q&A]
Has anyone made a process for deleting a bucket that isn't required with compliance mode objects?

Tried mc rb --force without success. I suppose we can stop minio and remove the files for the drives but I don't think that is a scalable solution.

Glen

**Comment by klauspost** [comment]
> In compliance mode, a protected object version can't be overwritten or deleted by any user, including the root user in your AWS account. When an object is locked in compliance mode, its retention mode can't be changed, and its retention period can't be shortened. Compliance mode helps ensure that an object version can't be overwritten or deleted for the duration of the retention period.

Being able to delete files in compliance mode within the retention period would defeat the purpose of it.

https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-lock-overview.html

**Comment by Jazz9** [comment]
@klauspost - Thankyou for the confirmation.
